### DISCUSSION TITLE
Move core functionality into a separate module e.g. spring-boot-htmx

### DISCUSSION THREAD

**Discussion by xhaggi** [wimdeblauwe/htmx-spring-boot#63] [Ideas]
Most of the functionality is independent from the template engine Thymeleaf. But if you want to use this module with another template engine like [freemarker](https://freemarker.apache.org/) or [jte](https://jte.gg/) you have to disable the auto configuration `HtmxThymeleafConfiguration` and exclude the `spring-boot-starter-thymeleaf` dependency to get it working. So I'm wondering if you should move the core functionality into a separate module e.g. `spring-boot-htmx` so that projects without Thymeleaf can also benefit from it. If you need some help here, let me know.

**Comment by wimdeblauwe** [comment]
I am open for this. @dsyer I believe you also had thoughts about this.

I am mostly wondering how to do it exactly. We would probably need to turn this project into a multi-module project and release different jar files. Not sure how to do that exactly.

**Comment by xhaggi** [comment]
>We would probably need to turn this project into a multi-module project and release different jar files. Not sure how to do that exactly.

I did it in #64. Let's move the discussion there please.
